### PR TITLE
Added auth_store.py manual code review and some changes

### LINKED ISSUE #71: Manual Code Review 2
Review HA Repos for any specific files related to your Misuse/Use cases

### LINKED ISSUE #77: Summary of Key Findings 2
Review a portion of the Key findings for the Github repo

### PR DESCRIPTION
**Pull request by CernickyR**

### CODE REVIEW

File: Code Analysis.md [hunk 1]
```diff
@@ -37,6 +37,11 @@ The rational behind the decisions are the following:
...
+
+[auth/auth_store.py](https://github.com/home-assistant/core/blob/dev/homeassistant/auth/auth_store.py)
+The auth_store.py file in the 'auth/' folder is the class which handles authenticaion for the Home Assistant server iteslf. It is responsible for retrieving/storing the authentication information for users of the system. Becuase authentication is an important aspect of any system and [CWE-287]( https://cwe.mitre.org/data/definitions/287.html) is on the Top 25 Most Dangerours Software Weakness List for 2020 is why we decided to do a manual code review of this file. Manually looking through the file they seem to hit most of the common authenticaion checks (i.e : verifying state of users account, having a is_owner group and is_user group seperated). One potential security flaw that may bring risk deals with the refresh tokens from line 428 to 454. This group of code seems to deasl with old refersh tokens used in the 'pre-0.78' version of the software. If any known vulnerabilities become relevant with that version, this legacy code check may carry those same risks along with it.
```
Comment (Chrs987) [inline Code Analysis.md]: Software Weakness List for 2020 is why we decided - remove is why and just have we decided
This group of code seems - change group to block

File: Code Analysis.md [hunk 2]
```diff
@@ -37,6 +37,11 @@ The rational behind the decisions are the following:
...
 
+ * [auth/auth_store.py](https://github.com/home-assistant/core/blob/dev/homeassistant/auth/auth_store.py)
```
Comment (Chrs987) [inline Code Analysis.md]: Move this bullet point up top to where the rest of them are. If it has issues I can just fix master once this is approved.

**Review by Chrs987 — DISMISSED** [review]
Just clean up the 2 comments and this will be good to go. You brought up a valid point for the code block on line 428-454.  This is not enabled by default as an authentication option so the risk is minimal but still there.

**Review by CernickyR — COMMENTED** [review]
changes

**Review by RocketSockz — APPROVED** [review]

**Review by Chrs987 — APPROVED** [review]
Thanks for making the changes.

**Pull request merged**